### DISCUSSION TITLE
How can I use ArtCNN vsmlrt version?

### DISCUSSION THREAD

**Discussion by ghost** [hooke007/mpv_PlayKit#471] [z_ENG_discuss]
How can I use ArtCNN vsmlrt version?

**Comment by hooke007** [comment]
Why not use ArtCNN's official ones instead?
https://github.com/Artoriuz/ArtCNN/tree/main/ONNX

I haven't check more.
It seems chroma variants are not supported by uai interface. Other variants should work.

**Reply by ghost** [reply]
I do not know how to use ONNX with mpv_lazy's k7sfunc

**Reply by hooke007** [reply]
I see you asked the similar question. https://github.com/hooke007/MPV_lazy/discussions/447

To make it simple if you do not know how to understand the doc in wiki.
Use the example script in repo.
https://github.com/hooke007/MPV_lazy/blob/main/portable_config/vs/ETC_UAI_NV_TRT.vpy
https://github.com/hooke007/MPV_lazy/blob/main/portable_config/vs/ETC_UAI_DML.vpy

Models should be placed in the model folder.

**Reply by ghost** [reply]
thank you so much!

**Reply by ghost** [reply]
@hooke007 I have a 1080p monitor, are these values good?
```
H_Pre = 0
Model = "ArtCNN_C16F64.onnx"
Gpu = 0
Gpu_T = 3
Res_Opt = [1280, 720]
Res_Max = [1920, 1200]
Ws_Size = 0
H_Max = 1080
Lk_Fmt = True
```

**Reply by hooke007** [reply]
I cannot talk exactly since I do not know what's your device and what u really want.

**Reply by ghost** [reply]
@hooke007 I have rtx 3050 and I want to upscale any lowerr res to 1080 or close to 1080p

**Reply by hooke007** [reply]
`Gpu_T = 2`

If most files' res is 1280x720 ...
`Res_Opt = [1280, 720]`

**Reply by Kytael** [reply]
I tried to use ArtCNN_C16F64 from the repo you linked using the UAI_NV_TRT.vpy template, and I got this error: 

`[I] Set shape of input tensor input for optimization profile 0 to: MIN=1x3x64x64 OPT=1x3x1080x1920 MAX=1x3x1440x2560
[E] Error[4]: [network.cpp::nvinfer1::Network::validate::3687] Error Code 4: Internal Error (input: for dimension number 1 in profile 0 does not match network definition (got min=3, opt=3, max=3), expected min=opt=max=1).)`

any idea what this error is?

**Reply by hooke007** [reply]
It's a greyscale model instead, which is out of my expectation. The wrapper needs update.

**Comment by hooke007** [comment]
It's much heavier than my expectation. But not reached the expected quality than I thought.
Added the new wrapper now. (example : https://github.com/hooke007/MPV_lazy/blob/main/portable_config/vs/SR_ARTCNN_NV.vpy )
